crates/extensions/az_mapper_ext/src/converter.rs

```diff
@@ -95,18 +104,20 @@ impl AzureConverter {
     }
 
     fn with_timestamp(&mut self, input: &MqttMessage) -> Result<String, ConversionError> {
-        let default_timestamp = self.add_timestamp.then(|| self.clock.now());
-        let mut payload_json: Map<String, Value> =
-            serde_json::from_slice(input.payload.as_bytes())?;
-
-        if let Some(timestamp) = default_timestamp {
-            let timestamp = timestamp
-                .format(&time::format_description::well_known::Rfc3339)?
-                .as_str()
-                .into();
-            payload_json.entry("time").or_insert(timestamp);
+        let default_timestamp = self.add_timestamp.then(|| IsoOrUnix::new(self.clock.now()));
+        let mut payload: Map<String, Value> = serde_json::from_slice(input.payload.as_bytes())?;
+
+        let time = payload
+            .remove("time")
+            .map(|value| IsoOrUnix::reformat_value(value, self.mapper_config.time_format))
```

Comment:
The actual code has changed to use a match instead now, but worth noting that annoyingly Rust doesn't allow you to specify a bound method as a closure like that, unlike Python, so it would be `.map(|val| self.mapper_config.time_format.format(val))`.